op-e2e/faultproof_test.go

```diff
@@ -290,6 +290,54 @@ func TestCannonDefendStep(t *testing.T) {
 	game.LogGameData(ctx)
 }
 
+func TestCannonChallengeWithCorrectRoot(t *testing.T) {
+	t.Skip("Not currently handling this case as the correct approach will change when output root bisection is added")
+	InitParallel(t)
+
+	ctx := context.Background()
+	sys, l1Client := startFaultDisputeSystem(t)
+	t.Cleanup(sys.Close)
+
+	l1Endpoint := sys.NodeEndpoint("l1")
+	l2Endpoint := sys.NodeEndpoint("sequencer")
+
+	disputeGameFactory := disputegame.NewFactoryHelper(t, ctx, sys.cfg.L1Deployments, l1Client)
+	game, correctTrace := disputeGameFactory.StartCannonGameWithCorrectRoot(ctx, sys.RollupConfig, sys.L2GenesisCfg, l1Endpoint, l2Endpoint, func(c *config.Config) {
+		c.TxMgrConfig.PrivateKey = e2eutils.EncodePrivKeyToString(sys.cfg.Secrets.Mallory)
+	})
+	require.NotNil(t, game)
+	game.LogGameData(ctx)
+
+	game.StartChallenger(ctx, sys.RollupConfig, sys.L2GenesisCfg, l1Endpoint, l2Endpoint, "Challenger", func(c *config.Config) {
+		c.AgreeWithProposedOutput = true // Agree with the proposed output, so disagree with the root claim
+		c.TxMgrConfig.PrivateKey = e2eutils.EncodePrivKeyToString(sys.cfg.Secrets.Alice)
+	})
+
+	maxDepth := game.MaxDepth(ctx)
+	for claimCount := int64(1); claimCount < maxDepth; {
+		game.LogGameData(ctx)
+		claimCount++
+		// Wait for the challenger to counter
+		game.WaitForClaimCount(ctx, claimCount)
+
+		// Defend everything because we have the same trace as the honest proposer
+		correctTrace.Defend(ctx, claimCount-1)
```

Comment:
When running this test locally, it seems to be stuck on waiting for claim count 2. It is failing with a timeout in that step which as far as I can tell, not what should be happening.

Reply:
That's exactly what we expect would happen at this point. The challenger agrees with the root claim so doesn't want to attack, but you can't defend the root claim either so it does nothing.